framework/wazuh/core/indexer/credential_manager.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+"""
+Keystore Client - SizeHeaderProtocol format
+"""
+
+import socket
+import json
+import struct
+
+SOCKET_PATH = "/var/ossec/queue/sockets/keystore"
+
+class KeystoreClient:
```

Comment:
Implementation of communication between Wazuh processes by queues is already implemented  at framework/wazuh/core/wazuh_socket.py